ppinot4py/computers/metrics_computer.py

```diff
@@ -131,35 +141,74 @@ def _first_to_after_from(from_condition, to_condition, id_values, ts_values):
     # Returns the first of them
     return df.groupby('id')['t'].first()
 
-def _linear_calculation(fromValues, toValues):
+def _linear_calculation(fromValues, toValues, measure):
     
-    finalResult = toValues - fromValues   
-    finalResult[finalResult < pd.Timedelta(0)] = pd.Timedelta('nan')
+    # Hour, min and second of start and close hour
+
+    if(measure.business_duration != None):
+        open_time= measure.business_duration.business_start
```

Comment:
You may want to move the code from here to line 173 so that you can reuse it in the cyclic measures